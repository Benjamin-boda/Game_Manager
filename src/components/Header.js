import React, {useContext} from "react";
import {Link, NavLink} from "react-router-dom";
import {useDispatch} from "react-redux";
import SearchGames from "./SearchGames";
import {startLogout} from "../actions/auth";
import { AuthContext } from "../firebase/Auth";


export const Header = () => {
    const dispatch = useDispatch()
    const {currentUser} = useContext(AuthContext);

    return (
        <div className="header">
            <div className="header__content">
                <div className="header__content__title">
                    <Link className="header__link" to="/homepage">
                        <h1 className="header__title">Game Manager</h1>
                    </Link>
                    <p className="header__paragraphe show-for-desktop">List the games you have to finish !</p>
                </div>              
                {!!currentUser ? <NavLink className="header__link" to="/dashboard">My Games</NavLink> : undefined}
                {!!currentUser ? <SearchGames/> : undefined}
                <a className="header__link show-for-desktop" href="https://github.com/Benjamin-boda?tab=repositories">Github</a>
                {!!currentUser ? <button className="button header__link header__logout__mobile" onClick={() => {dispatch(startLogout())}}>Logout</button> : undefined}
            </div>
        </div>
    )
}

export default Header;