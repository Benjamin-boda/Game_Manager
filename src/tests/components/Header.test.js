import { mount } from "enzyme";
import Header from "../../components/Header";
import {AuthProvider} from "../../firebase/Auth";
import { Provider, useSelector, useDispatch } from "react-redux";
import React, {useContext} from "react";
import thunk from 'redux-thunk'
import configureMockStore from "redux-mock-store";
import { shallowToJson } from "enzyme-to-json";
import { MemoryRouter } from 'react-router-dom'

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

test("should render Header page correctly", () => {
    
    const wrapper = shallowToJson(mount(
        <Provider store={store}>
            <AuthProvider >
            <MemoryRouter  initialEntries={['/']} keyLength={0}>
            <Header 
                /> 
            </MemoryRouter>
            </AuthProvider>
                
        </Provider>
        ))
    expect(wrapper).toMatchSnapshot();
});