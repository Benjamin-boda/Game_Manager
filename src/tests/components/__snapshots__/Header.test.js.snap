// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header page correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <AuthProvider>
    <MemoryRouter
      initialEntries={
        Array [
          "/",
        ]
      }
      keyLength={0}
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Header>
          <div>
            <Link
              to="/homepage"
            >
              <LinkAnchor
                href="/homepage"
                navigate={[Function]}
              >
                <a
                  href="/homepage"
                  onClick={[Function]}
                >
                  <h1>
                    Game Manager
                  </h1>
                </a>
              </LinkAnchor>
            </Link>
            <p>
              List the games you have to finish !
            </p>
            <SearchGames>
              <div>
                <input
                  onChange={[Function]}
                  placeholder="Search a game..."
                  type="text"
                  value=""
                />
                <div>
                  <ul>
                    <SearchGamesDropDownNoItem>
                      <li>
                        <span>
                          No result
                        </span>
                      </li>
                    </SearchGamesDropDownNoItem>
                  </ul>
                </div>
              </div>
            </SearchGames>
          </div>
        </Header>
      </Router>
    </MemoryRouter>
  </AuthProvider>
</Provider>
`;
